src/SelectPanel/SelectPanel.tsx

```diff
@@ -72,6 +100,13 @@ export function SelectPanel({
     [externalOnFilterChange, setInternalFilterValue]
   )
 
+  const selectedItems = React.useMemo(
+    () => (Array.isArray(selected) ? selected : [...(selected ? [selected] : [])]),
+    [selected]
+  )
+
+  const [finalItemsSelected, setFinalItemsSelected] = useState(selectedItems)
```

Comment:
@bolonio I think I found the issue that we were running into with the selected state and why it wasn't reproducing well in Storybook. In our app, we're passing in the selected state immediately and then asynchronously loading all of the options. But this selected state inside `SelectPanel` doesn't get reset when that happens, so it appears that nothing is initially selected. I was able to fix that by patching our `@primer/react` built file to have this line:

```suggestion
  const [finalItemsSelected, setFinalItemsSelected] = useState(selectedItems)
  React.useEffect(() => setFinalItemsSelected(selectedItems), [selectedItems])
```

I think partially why our code worked before was because there was no `useState` here, so the state was automatically reset on every render, which is not true anymore.

Reply:
Thank you for debugging it! I'll add the fix and reproduce this use case in one of the stories.